### PR TITLE
[Python] Adds new client generator, python-experimental

### PR DESCRIPTION
**Pull request by spacether**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`. If contributing template-only or documentation-only changes which will change sample output, be sure to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) first.
- [X] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [X] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

@taxpon (2017/07) @frol (2017/07) @mbohlool (2017/07) @cbornet (2017/09) @kenjones-cisco (2017/11) @tomplus (2018/10) @Jyhess (2019/01)

### Description of the PR

This creates the the `python-experimental` generator which is a client generator based on the `python` generator. 

This was requested by the maintainers as a place to continue work on:
- adding models for objects whose type is != objects which have enums and validations that we want to check
- automatically loading enums of length one into model and endpoint required parameters
- class based typing in python
- additionalProperties on object models

If this is merged, I will merge the following PRs into it:
- https://github.com/OpenAPITools/openapi-generator/pull/3118
- https://github.com/OpenAPITools/openapi-generator/pull/2757

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java]: @spacether for new files, we don't need line 3 as per https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license/7301#7301

File: samples/client/petstore/perl/t/AnotherFakeApiTest.t [hunk 2]
```diff
@@ -0,0 +1,41 @@
+=begin comment
```
Comment (wing328) [inline samples/client/petstore/perl/t/AnotherFakeApiTest.t]: I don't think this file should be included ...

We've cleared this issue (corrupted file?) in the latest master.

Comment (spacether) [reply samples/client/petstore/perl/t/AnotherFakeApiTest.t]: This file has been removed

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java (hunk 1 above)
Comment (spacether) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java]: The line ` * Copyright 2018 SmartBear Software` has been removed

**Comment by spacether** [comment]
@wing328 all feedback has been addressed

**Comment by wing328** [comment]
@spacether can you please also copy the python test and update pom.xml to test the python (experimental) petstore sample?

(I've done something similar for Go: https://github.com/OpenAPITools/openapi-generator/pull/3268)

**Comment by spacether** [comment]
> @spacether can you please also copy the python test and update pom.xml to test the python (experimental) petstore sample?
> 
> (I've done something similar for Go: #3268)

@wing328 
- the existing python tests have been copied into python-experimental
- pom.xml has been updated so python-experimental sample tests run in Travis
- all tests pass

**Comment by wing328** [comment]
For your next PR, you may want to add a windows batch file to generate the Python experimental petstore client and contributors using Windows will find it easier to update the samples.

**Pull request merged**